pkg/kgo/config.go

```diff
@@ -1536,6 +1544,22 @@ func ConsumeRegex() ConsumerOpt {
 	return consumerOpt{func(cfg *cfg) { cfg.regex = true }}
 }
 
+// ConsumeExcludeTopics sets topics to exclude when using regex consumption.
+// This option only has effect when ConsumeRegex is enabled.
+//
+// Topics matching any of the provided regular expressions will be excluded from
+// consumption, even if they match patterns provided to ConsumeTopics.
+func ConsumeExcludeTopics(topics ...string) ConsumerOpt {
```

Comment:
Alternatively another API since kafka topics cannot have the character `!` in them you could adapt something like this in the existing pattern and interpret them as negative patterns:

https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax#example-including-and-excluding-branches

Reply:
Yes.

I think the way it is it's easier to understand for users, and is self documenting.